org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/internal/ZigBeeNetworkStateSerializerImpl.java

```diff
@@ -145,9 +165,19 @@ public void deserialize(final ZigBeeNetworkManager networkState) {
 
         final File file = getNetworkStateFile();
         boolean networkStateExists = file.exists();
+
         if (!networkStateExists) {
-            logger.debug("Loading ZigBee network state: File does not exist");
-            return;
+            final File temp = new File(file.getAbsolutePath() + ".new");
+            if (!temp.exists()) {
+                logger.debug("Loading ZigBee network state: File does not exist");
+                return;
+            }
+
+            logger.warn("Recovering network state file from temporary copy");
+            if (!temp.renameTo(file)) {
+                logger.error("Error recovering network state file");
```

Comment:
As before,arrived at this point, we also lost the network state file. It was renamed, and renaming it back to the the original name failed even if we know the target file does not exist: there is something really wrong with the setup here.
I would keep it as error.

Reply:
I don't really agree - what does the user do?  Do you think it fits the criterea?

> error logging should only be used to inform the user that something is tremendously wrong in his setup, the system cannot function normally anymore

Since the system will normally recover after the loss of the file, I'm not sure this is correct.

Reply:
As above, if the files in the the config directory cannot be renamed, there is definetely something really wrong in the installation. And I was also trying to be consistent with the existing code, as if the file write fails due to an exception, it is reported as error. The only difference here is that file open / write / close operations launch an exception in case of error, while renames and deletes return false instead.

But I have anyway changed it to warn, as requested.

Reply:
>  I was also trying to be consistent with the existing code

Just because I used error in the past, doesn't mean it's right ;) I also have a bad habit of setting the levels too high - thinking about it like a programmer, rather than a user.